optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/uni/DroolsUniCondition.java

```diff
@@ -86,11 +90,59 @@ private DroolsUniCondition(DroolsMetadata<Object, A> aMetadata) {
         return new DroolsUniCondition<>(aMetadata.substitute(patternSupplier));
     }
 
+    public static Index.ConstraintType getConstraintType(JoinerType type) {
+        switch (type) {
+            case EQUAL:
+                return Index.ConstraintType.EQUAL;
+            case LESS_THAN:
+                return Index.ConstraintType.LESS_THAN;
+            case LESS_THAN_OR_EQUAL:
+                return Index.ConstraintType.LESS_OR_EQUAL;
+            case GREATER_THAN:
+                return Index.ConstraintType.GREATER_THAN;
+            case GREATER_THAN_OR_EQUAL:
+                return Index.ConstraintType.GREATER_OR_EQUAL;
+            default:
+                throw new IllegalStateException("Unsupported joiner type (" + type + ").");
```

Comment:
Can we afford to do this?

Reply:
Yes, we always have a similar default for all switch statements, because you never know someone adds a enum value without adjusting this code.

That being said, should we switch case all cases? Yes, we should, if this exception pops up, it's a priority issue. Unit testing should cover all enum cases.
I don't see another way of dealing with this.